samza-core/src/main/java/org/apache/samza/zk/ZkControllerImpl.java

```diff
@@ -83,12 +83,12 @@ public boolean isLeader() {
 
   @Override
   public void stop() {
-    if (isLeader()) {
-      zkLeaderElector.resignLeadership();
-    }
-
-    // close zk connection
-    if (zkUtils != null) {
+    try {
+      if (isLeader()) {
+        zkLeaderElector.resignLeadership();
+      }
+    } finally {
+      zkUtils.deleteProcessorNode(processorId);
```

Comment:
`zkUtils.deleteProcessorNode()` should internally catch all exceptions since it's on the finally block.